src/Http/Http/src/StreamPipeWriter.cs

```diff
@@ -278,7 +278,8 @@ private void AddSegment(int sizeHint = 0)
             }
 
             // Get a new buffer using the minimum segment size, unless the size hint is larger than a single segment.
-            _currentSegmentOwner = _pool.Rent(Math.Max(_minimumSegmentSize, sizeHint));
+            // Also, the size cannot be larger than the MaxBufferSize of the MemoryPool
+            _currentSegmentOwner = _pool.Rent(Math.Min(Math.Max(_minimumSegmentSize, sizeHint), _pool.MaxBufferSize));
```

Comment:
```suggestion
            _currentSegmentOwner = _pool.Rent(Math.Clamp(sizeHint, _minimumSegmentSize, _pool.MaxBufferSize));
```

https://docs.microsoft.com/en-us/dotnet/api/system.math.clamp?view=netcore-2.2#System_Math_Clamp_System_Int32_System_Int32_System_Int32_

Reply:
Yo that API is hype.